e3db/types/queryadv.py

```diff
@@ -0,0 +1,150 @@
+import time
+from .query_range import Range
+from .query_params import Params
+
+class QueryAdv(object):
+    def __init__(self, last_index=0, count=50, include_all_writers=False, include_data=False, match=None, exclude=None, range=None): 
+        """
+        Initialize the Queryv2 class.
+
+        This object type is used to build a v2/search Query to run against the server.
+        It is exposed through the e3db.Client.queryv2 method.
+
+        Parameters
+        """
+        self.__next_token = int(last_index)
+        self.__limit = int(count)
+        self.__include_data = bool(include_data)
+        self.__include_all_writers = bool(include_all_writers)
+        self.__match = match if match else [] 
+        self.__exclude = exclude if exclude else []
+        self.__range = None
+        
+    def append_match(self, p):
+        self.__match.append(p)
+
+    def append_exclude(self, p):
+        self.__exclude.append(p)
+
+    # count getters
+    @property
+    def count(self):
+        """
+        Get count of Query.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        int
+            Get count of records.
+        """
+        return self.__limit
+
+    # after_index getters
+    @property
+    def after_index(self):
+        """
+        Get after_index of Query.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        int
+            Get after_index of the Query.
+        """
+        return self.__next_token
+
+    @after_index.setter
+    def after_index(self, value):
+        """
+        Set after_index of QueryResult
+
+        Parameters
+        ----------
+        value: int
+            Set after_index from server result
+
+        Returns
+        -------
+        None
+        """
+        self.__next_token = int(value)
+
+    # include_data getters
+    @property
+    def include_data(self):
+        """
+        Get included_data of Query.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        bool
+            Whether data is included with the records.
+        """
+        return self.__include_data
+
+
+    # include_all_writers getters
+    @property
+    def include_all_writers(self):
+        """
+        Get whether the query included all writers.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        bool
+            include_all_writers
+        """
+        return self.__include_all_writers
+
+    def to_json(self):
```

Comment:
This isn't serializing to a json, it just converts to a dict. Which is totally fine but the naming could cause confusion.